### PR TITLE
Extended docs regarding ValueAtQuantile() and added ValuesAreEquivalent()

### PR DESCRIPTION
**Pull request by filipecosta90**
This PR introduces the ValuesAreEquivalent() method and extendes the ValueAtQuantile() docs so that it's clear that it returns the largest value that (100% - percentile) of the overall recorded value entries in the histogram are either larger than or equivalent to. 
It can be used as further reference for issue #23 and why it happens. 
wdyt @ahothan @giltene ?

### CODE REVIEW

File: hdr.go [hunk 1]
```diff
@@ -69,23 +69,32 @@ func (h *Histogram) SetStartTimeMs(startTimeMs int64) {
...
+// Construct a Histogram given the Lowest and Highest values to be tracked and a number of significant decimal digits.
+//
+// Providing a lowestDiscernibleValue is useful is situations where the units used for the histogram's values are
+// much smaller that the minimal accuracy required.
+// E.g. when tracking time values stated in nanosecond units, where the minimal accuracy required is a microsecond, the proper value for lowestDiscernibleValue would be 1000.
+func New(lowestDiscernibleValue, highestTrackableValue int64, numberOfSignificantValueDigits int) *Histogram {
+	if numberOfSignificantValueDigits < 1 || 5 < numberOfSignificantValueDigits {
+		panic(fmt.Errorf("numberOfSignificantValueDigits must be [1,5] (was %d)", numberOfSignificantValueDigits))
+	}
+
```
Comment (ahothan) [inline hdr.go:81]: add check on lowestDiscernibleValue must be >0
suggest forcing to 1 as min value

File: hdr.go [hunk 2]
```diff
@@ -69,23 +69,32 @@ func (h *Histogram) SetStartTimeMs(startTimeMs int64) {
...
-// range and with the given amount of precision.
-func New(minValue, maxValue int64, sigfigs int) *Histogram {
-	if sigfigs < 1 || 5 < sigfigs {
-		panic(fmt.Errorf("sigfigs must be [1,5] (was %d)", sigfigs))
-	}
-
-	largestValueWithSingleUnitResolution := 2 * math.Pow10(sigfigs)
+// Construct a Histogram given the Lowest and Highest values to be tracked and a number of significant decimal digits.
+//
+// Providing a lowestDiscernibleValue is useful is situations where the units used for the histogram's values are
```
[the remark this answers is not in the document]
Comment (filipecosta90) [reply hdr.go]: Fixed in the latest commit

File: hdr.go (hunk 1 above)
Comment (filipecosta90) [reply hdr.go:85]: @ahothan added the force to 1 as minimum discernible value. Regarding panic imho we should avoid extra panic calls within the code. Ideally we should deprecate `New()` in the future and enable a constructor that returns both the histogram and error. But for now I would stick to this constructor and only forcing 1 as a min value. wdyt?

Comment (ahothan) [reply hdr.go:85]: I agree we should never use panic inside a library.
I would suggest to also force the value within range and document accordingly - much better than panic.
Force to nearest boundary (1 if <1 or 5 if >5 would be appropriate).

**Review by ahothan — COMMENTED** [review]
Looks good, but could you please take care of removing the panic as commented.

File: hdr.go (hunk 1 above)
Comment (filipecosta90) [reply hdr.go:85]: Agree. doing the change :+1:

**Comment by filipecosta90** [comment]
> Looks good, but could you please take care of removing the panic as commented.

@ahothan changes per the PR review are in :)

File: hdr.go [hunk 3]
```diff
@@ -69,23 +69,42 @@ func (h *Histogram) SetStartTimeMs(startTimeMs int64) {
...
+// E.g. when tracking time values stated in nanosecond units, where the minimal accuracy required is a microsecond,
+// the proper value for lowestDiscernibleValue would be 1000.
+//
+// Note: the numberOfSignificantValueDigits must be [1,5]. If lower than 1 the numberOfSignificantValueDigits will be
+// forced to 1, and if higher than 5 the numberOfSignificantValueDigits will be forced to 5.
+func New(lowestDiscernibleValue, highestTrackableValue int64, numberOfSignificantValueDigits int) *Histogram {
+	if numberOfSignificantValueDigits < 1 {
+		numberOfSignificantValueDigits = 1
+	}
+	if numberOfSignificantValueDigits > 5 {
```
Comment (ahothan) [inline hdr.go:85]: (nitpick) else if

**Review by ahothan — APPROVED** [review]

File: hdr.go (hunk 3 above)
Comment (filipecosta90) [reply hdr.go]: @ahothan changed per the review. I've also added whitebox testing specifically for this conditions. 
can you check it?

**Comment by ahothan** [comment]
Looks good!

**Pull request merged**